parcllabs/services/parcllabs_service.py

```diff
@@ -1,142 +1,254 @@
+
+        Raises:
+            RequestException: If the request fails or an unexpected error occurs.
+        """
+        try:
+            if method.upper() == "GET":
+                params = kwargs.get("params", {})
+                if not params.get("limit"):
+                    params["limit"] = self.limit
+                kwargs["params"] = params
+
+            response = requests.request(method, url, headers=self.headers, **kwargs)
+            response.raise_for_status()
+            return response
+        except requests.exceptions.HTTPError:
+            self.error_handling(response)
+        except requests.exceptions.RequestException as err:
+            raise RequestException(f"Request failed: {str(err)}")
+        except Exception as e:
+            raise RequestException(f"An unexpected error occurred: {str(e)}")
+
+    def _post(
+        self, url: str, data: Optional[Dict[str, Any]] = None
+    ) -> requests.Response:
+        """
+        Send a POST request to the specified URL with the given data.
+
+        Args:
+            url (str): The URL endpoint to request.
+            data (dict, optional): The data to send in the request body.
+
+        Returns:
+            requests.Response: The response object.
+        """
+        return self._make_request("POST", url, json=data)
+
+    def _get(
+        self, url: str, params: Optional[Dict[str, Any]] = None
+    ) -> requests.Response:
+        """
+        Send a GET request to the specified URL with the given parameters.
+
+        Args:
+            url (str): The URL endpoint to request.
+            params (dict, optional): The parameters to send in the query string.
+
+        Returns:
+            requests.Response: The response object.
+        """
+        return self._make_request("GET", url, params=params)
+
+    def _fetch(
+        self,
+        parcl_ids: List[int],
+        params: Optional[Mapping[str, Any]],
+        auto_paginate: bool = False,
+    ):
+        """
+        Fetch data for given Parcl IDs with specified parameters.
+
+        This method handles the fetching of data based on the provided Parcl IDs and parameters.
+        It supports both GET and POST requests, depending on the client's configuration and the number of Parcl IDs.
+
+        Args:
+            parcl_ids (List[int]): A list of Parcl IDs to fetch data for.
+            params (Optional[Mapping[str, Any]]): Additional parameters for the request.
+                If not provided or None, no additional parameters will be used.
+            auto_paginate (bool, optional): Whether to automatically handle pagination.
+                Defaults to False.
+
+        Returns:
+            The result of the fetch operation. The exact return type depends on the specific
+            fetch method called (_fetch_post, _fetch_get, or _fetch_get_many_parcl_ids).
+        """
+        if params and not params.get("limit"):
+            params["limit"] = self.limit
```

Comment:
this seems to be repeated in `clean_params` also, we can probably get rid of this one or the other one

Reply:
Great flag - killed it.